src/backend/marsha/core/tests/test_api_video_live_pairing.py

```diff
@@ -0,0 +1,401 @@
+        """Throttling should prevent more than three requests per minute.
+
+        Possible scenarios:
+        - Multiple requests from legit box that were never been paired.
+        - Brute force attempt to guess a secret.
+        """
+        faker = Faker()
+        payload = {"box_id": uuid.uuid4(), "secret": "000000"}
+
+        # first 3 requests shouldn't be throttled
+        for _ in range(3):
+            response = self.client.post(
+                "/api/pairing-challenge",
+                data=payload,
+                REMOTE_ADDR=faker.ipv4(),
+            )
+            self.assertEqual(response.status_code, 404)
+            self.assertEqual(response.json(), {"detail": "Secret not found."})
+
+        # fourth request should be throttled
+        response = self.client.post(
+            "/api/pairing-challenge",
+            data=payload,
+            REMOTE_ADDR=faker.ipv4(),
+        )
+        self.assertEqual(response.status_code, 429)
+        self.assertEqual(
+            response.json(),
+            {"detail": "Request was throttled. Expected available in 60 seconds."},
+        )
+
+        # resetting throttling by mocking timer used by DRF in AnonRateThrottle
+        with mock.patch.object(time, "time", return_value=time.time() + 60):
+            # first 3 requests shouldn't be throttled
+            for _ in range(3):
+                response = self.client.post("/api/pairing-challenge", data=payload)
+                self.assertEqual(response.status_code, 404)
+                self.assertEqual(response.json(), {"detail": "Secret not found."})
+
+            # fourth request should be throttled
+            response = self.client.post("/api/pairing-challenge", data=payload)
+            self.assertEqual(response.status_code, 429)
+            self.assertEqual(
+                response.json(),
+                {"detail": "Request was throttled. Expected available in 60 seconds."},
+            )
+
+    def test_api_video_pairing_challenge_unknown_device_throttling(self):
+        """Unknown device requests should be throttled with others."""
+        box_id = uuid.uuid4()
+        payload = {"box_id": box_id, "secret": "000000"}
+
+        # first 3 requests shouldn't be throttled
+        for _ in range(3):
+            response = self.client.post("/api/pairing-challenge", data=payload)
+            self.assertEqual(response.status_code, 404)
+            self.assertEqual(response.json(), {"detail": "Secret not found."})
+
+        # fourth known box request should be throttled
+        response = self.client.post("/api/pairing-challenge", data=payload)
+        self.assertEqual(response.status_code, 429)
+        self.assertEqual(
+            response.json(),
+            {"detail": "Request was throttled. Expected available in 60 seconds."},
+        )
+
+        # other request should be throttled
+        response = self.client.post(
```

Comment:
Can  you explain why it was throttled since you configured the throttle to check the box ID? My understanding is that a new box id should be throttled independently of the first one.

Reply:
This was indeed the case, but I changed this recently :
If the box is known, the throttling uses the box id as a key.
Otherwise, no key is set, so every unknown request shares the same throttling rate.